paz/models/detection/frustum_pointnet.py

```diff
@@ -0,0 +1,177 @@
+
+    net = Conv2D(512, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_6')(
+        concat_feat)
+
+    net = Conv2D(256, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_7')(net)
+
+    net = Conv2D(128, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_8')(net)
+
+    net = Conv2D(128, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_9')(net)
+
+    net = Dropout(rate=0.5)(net)
+
+    logits = Conv2D(2, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation=None, name='conv1_10')(net)
+
+    logits = tf.squeeze(logits, [2])  # BxNxC
+
+    return logits
+
+
+def Box_Est_Net(object_point_cloud, one_hot_vec):
+    num_point = object_point_cloud.get_shape()[1]
+    net = tf.expand_dims(object_point_cloud, 2)
+
+    net = Conv2D(128, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv2_1')(net)
+
+    net = Conv2D(128, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv2_2')(net)
+
+    net = Conv2D(256, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv2_3')(net)
+
+    net = Conv2D(512, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv2_4')(net)
+
+    net = MaxPooling2D(pool_size=[num_point, 1], padding='VALID')(net)
+
+    net = tf.squeeze(net, axis=[1, 2])
+    net = tf.concat([net, one_hot_vec], axis=1)
+
+    net = Dense(units=512, activation='relu')(net)
+    net = Dense(units=256, activation='relu')(net)
+
+    output = Dense(units=3 + NUM_HEADING_BIN * 2 + NUM_SIZE_CLUSTER * 4, activation=None)(net)
+
+    return output
+
+
+def TNet(object_point_cloud, one_hot_vec):
+    num_point = object_point_cloud.get_shape()[1]
+    net = tf.expand_dims(object_point_cloud, 2)
+
+    net = Conv2D(128, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv3_1')(net)
+
+    net = Conv2D(128, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv3_2')(net)
+
+    net = Conv2D(256, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv3_3')(net)
+
+    net = MaxPooling2D(pool_size=[num_point, 1], padding='VALID')(net)
+
+    net = tf.squeeze(net, axis=[1, 2])
+    net = tf.concat([net, one_hot_vec], axis=1)
+
+    net = Dense(units=256, activation='relu')(net)
+    net = Dense(units=128, activation='relu')(net)
+    predicted_center = Dense(units=3, activation=None)(net)
+
+    return predicted_center
+
+
+def FrustumPointNetModel(point_cloud_shape=(1024, 3), one_hot_vec_shape=(3,), mask_label_shape=(1024,),
+                         center_label_shape=(3,), heading_class_label_shape=(), heading_residual_label_shape=(),
+                         size_class_label_shape=(), size_residual_label_shape=(3,), batch_size=32):
+    end_points = {}
```

Comment:
"end_points" is not a clear variable name. If it's meant to hold the network outputs then name it "network_outputs" or "model_outputs" or "outputs" instead.

Reply:
Done